circuit_breaker_test.go

```diff
@@ -0,0 +1,259 @@
+package aibridge
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/sony/gobreaker/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCircuitBreakerMiddleware_TripsOnUpstreamErrors(t *testing.T) {
```

Comment:
This test seem to mostly test library not aibridge code, maybe verify metrics here. In general metric values should be tested somewhere.

Would be nice to have more of "integration" test that would check handlers in `NewRequestBridge` + provider configuration.